src/receipt.rs

```diff
@@ -182,8 +185,20 @@ impl Receipt {
         }
     }
 
-    pub const fn panic(id: ContractId, reason: InstructionResult, pc: Word, is: Word) -> Self {
-        Self::Panic { id, reason, pc, is }
+    pub const fn panic(
+        id: ContractId,
+        reason: InstructionResult,
+        pc: Word,
+        is: Word,
+        contract_id: Option<ContractId>,
```

Comment:
Do we need this field here? Maybe it is better to leave `None` by default and set it in the places where we can do that. It will avoid a lot of `None` in tests=)

Reply:
Now the value is `None` by default, and we can change it using `set_panic_contract_id`